### ISSUE TITLE
More consistent handling of 'None' vs. 'none'

### ISSUE THREAD

**Issue by timhoffm** [matplotlib/matplotlib#19300]
**Summary**

'None' vs 'none' is not handled consistently:

- At least MarkerStyle only accepts 'None' (xref https://github.com/matplotlib/matplotlib/pull/19291#discussion_r557735785). - Maybe other places too.
- However, in many places we accept both 'None', 'none' (and sometimes also other capitalizations).
- Counting occurences, we have 777 'none' and 139 'None' in our code base.

### Proposed fix

Aim at making the API more consistent by:

- Choose one version and use it whenever possible throughout code, docs and examples.
  - I propose to choose 'none' because most other named strings are lowercase. And it's a step further away from `None`, which may help making it more clear that those two are different.
- Make all other places accept that version too.
- For now do **not** deprecate other writings or conversions. If a way of writing is working now, there'll be lots of code out there using it. It's not worth breaking that.

**Comment by mwaskom** [comment]
Aren't there places where they mean different things? i.e. `None` means "use the rcParams default and `"none"` means "use an appropriate null value"?

This is the case at very least for `facecolor`, probably others...

**Comment by timhoffm** [comment]
This is about capitalization of strings `'None'` vs. `'none'`. You are right that `None` is something completely different.

**Comment by mwaskom** [comment]
Oops! Missed that entirely.

**Comment by brunobeltran** [comment]
I have definitely encountered this in several places (including markers). For now I've been erring on the side of accepting both `"None"` and `"none"` as synonyms if they already were used before, but only documenting that we accept `'none'`. Sounds like this is completely in line with your proposal.

I'd add my vote for `'none'` to also be added in places where it is not currently accepted, and documentation updated.

**Comment by oscargus** [comment]
I was playing around with this and noted two things:
1. line styles are converted into `"None"` as the preferred representation:
https://github.com/matplotlib/matplotlib/blob/3699ff34d6e2d6d649ee0ced5dc3c74936449d67/lib/matplotlib/lines.py#L1112-L1113
2. Some code relies on that:
https://github.com/matplotlib/matplotlib/blob/3699ff34d6e2d6d649ee0ced5dc3c74936449d67/lib/matplotlib/axes/_base.py#L474-L477

**Comment by timhoffm** [comment]
Yes it’s a bit of a mess.

**Comment by timhoffm** [comment]
While there has been progress, this topic is not completely solved.

**Comment by anntzer** [comment]
Along similar lines (though not exactly the same issue), something that bit me recently is that I think the way to check that a line has no marker set is currently something like `line.get_marker() in ["None", "none", " ", "", None]`, which is not so great.

**Comment by timhoffm** [comment]
Agreed. How do we get out of there? First, do we have a policy / opinion on whether we prefer roundtrips (i.e. `line.set_marker(m); line.get_marker() == m` is always true) or normalization (i.e. `line.get_marker()` for empty markers is always `"none"`)? Similar related discussion on linestyle: https://github.com/matplotlib/matplotlib/pull/29304

Second, if we prefer normalization, that'd be a breaking change. At least usage seems quite low:
https://github.com/search?q=%2F%5C.get_marker%5C%28%5C%29+%3D%3D+%5B%22%27%5D%28%5Cs%3F%7Cnone%29%5B%22%27%5D%2F+language%3APython+NOT+is%3Afork+NOT+path%3A**%2Fmatplotlib%2F**+NOT+path%3A**%2Fsite-packages%2F**&type=code&p=4

**Comment by anntzer** [comment]
I guess I'd prefer normalization and would be willing to sacrifice roundtripping.

**Comment by timhoffm** [comment]
I'm also inclined that way: There's one canonical name. Other spellings are accepted as a convenience, but they are normalized early and not used interernally or returned.

Can we afford a hard break on the return side? Or do we need a deprecation song-and-dance, which likely means we warn if `get_marker()` returns one of the alternative spellings and tell users that they should ideally adapt their inputs, or at least make the code using the return value compatible with "none".
